devito/types/dimension.py

```diff
@@ -615,10 +615,21 @@ def _offset_right(self):
     def _arg_defaults(self, grid=None, **kwargs):
         if grid is not None and grid.is_distributed(self.root):
             # Get local thickness
-            ltkn = grid.distributor.glb_to_loc(self.root,
-                                               self.thickness.left[1], LEFT) or 0
-            rtkn = grid.distributor.glb_to_loc(self.root,
-                                               self.thickness.right[1], RIGHT) or 0
+            if self.local:
+                # dimension is of type ``left``/right`` - compute the 'offset'
+                # and then add 1 to get the appropriate thickness
+                ltkn = grid.distributor.glb_to_loc(self.root,
+                                                   self.thickness.left[1]-1, LEFT)
```

Comment:
so this removes one (-1) from thickness and add it back below? looks weird.

Reply:
the `glb_to_loc` is computing the offset, not the thickness. Consider the following example: we have a grid:
```
0-1-2-3-4
```
Then, i.e. `glb_to_loc(d, 0, LEFT)` would return 0. `glb_to_loc(d,4,LEFT)` would return 4 and `glb_to_loc(d,5,LEFT)` would return `None` since it's OOB. Hence we take the 1 off to get the offset but then we need to add it back to get the thickness (as defined for subdimensions of type `left`/`right` which have a minimum 'thickness' of 1).

Reply:
Ok makes sense.